src/types.ts

```diff
@@ -0,0 +1,46 @@
+import { Options } from './index';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// TODO: define what a config is ?
+// interface ResolvedConfig {
+//   [key: string]: unknown;
+// }
+// type ConfigFn = (...params: any[]) => ResolvedConfig;
+// type ConfigFnPromise = (...params: any[]) => Promise<ResolvedConfig>;
+// type ConfigPromise = Promise<ResolvedConfig>;
+
+// https://github.com/davidtheclark/cosmiconfig/issues/134
+export type Config = any;
+// | ResolvedConfig
+// | ConfigFn
+// | ConfigPromise
```

Comment:
Is it expected for a user to be able to remove a file type loader by providing an `undefined`/`null` value value? If so I'll need to add `void` here. (I'm not requesting this, just want to have the types be correct)